airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -216,7 +216,16 @@ def _get_conn_params(self) -> dict[str, str | None]:
                 "Please remove one."
             )
         elif private_key_file:
-            private_key_pem = Path(private_key_file).read_bytes()
+            private_key_file_path = Path(private_key_file)
+            if not private_key_file_path.is_file() or private_key_file_path.stat().st_size == 0:
+                raise ValueError(
+                    "The private_key_file path points to an empty or invalid file."
+                )
+            if private_key_file_path.stat().st_size > 4096:
+                raise ValueError(
+                    "File size of private_key_file is too big."
```

Comment:
```suggestion
                    "The private_key_file size is too big. Please keep it less than 4 KB"
```

Reply:
Let's also add a note in the docs about the size limitation.

Reply:
What documentation do you suggest putting it to? The limitations is applied to the connection settings. I coulad add a note to the operator docs.